src/Microsoft.AspNetCore.Mvc.Razor/Compilation/ViewsFeatureProvider.cs

```diff
@@ -27,20 +27,37 @@ public class ViewsFeatureProvider : IApplicationFeatureProvider<ViewsFeature>
         /// <inheritdoc />
         public void PopulateFeature(IEnumerable<ApplicationPart> parts, ViewsFeature feature)
         {
-            foreach (var assemblyPart in parts.OfType<AssemblyPart>())
+            var knownItems = new Dictionary<string, (AssemblyPart part, string identifier)>(StringComparer.OrdinalIgnoreCase);
+            foreach (var assemblyPart in parts.OfType<AssemblyPart>().OrderBy(part => part.IsEntryPoint ? 0 : 1))
             {
                 var attributes = GetViewAttributes(assemblyPart);
                 var items = LoadItems(assemblyPart);
 
                 var merged = Merge(items, attributes);
-                foreach (var entry in merged)
+                foreach (var kvp in merged)
                 {
-                    feature.ViewDescriptors.Add(new CompiledViewDescriptor(entry.item, entry.attribute));
+                    var identifier = kvp.Key;
```

Comment:
We need a way to Opt-In to allow libraries to opt-in for their precompiled views to get picked up by the system.
When it comes to libraries we can't assume that it's OK to just plug in the views (and expose the endpoints). As those views might need to be configured during the startup sequence (for example to add authentication)

So we should provide a way for libraries to have their pre-compiled views/pages disabled by default and plugged in through a gesture that gives them an opportunity to fully configure the endpoints before exposing them)